### DISCUSSION TITLE
Function aggregation feature

### DISCUSSION THREAD

**Discussion by lakylekidd** [dotnet/csharplang#4751] [Language Ideas]
Hello All,

I was thinking of a cool implementation for CSharp which involves aggregating functions. Basically, we come across many situations where we build a wrapper service/contract with several functions that all point to the same internal one, just so we can define a standard contract for future extension. Example:

```
public void DoSomething()
{
    this.SubFunction();
}

public void DoSomethingElse()
{
    this.SubFunction();
}
```

It would be cool if we could aggregate these parent functions, just like a switch function like so:

```
public void DoSomething()
public void DoSomethingElse()
{
   this.SubFunction();
}
```

What do you guys think? Would that be something useful? Can you think of a different implementation for this functionality? 

I'd love to hear your thoughts,
Billy

**Comment by svick** [comment]
Personally, I don't see how this kind of syntax sugar could raise to the level where it would be worthwhile to add it to the language.

Consider:

* It's not something that would be used often.
* It does not save much code, especially since you could use expression-bodied methods here:

    ```c#
    public void DoSomething() => this.SubFunction();
    public void DoSomethingElse() => this.SubFunction();
    ```
* I think it does not make the code clearer.

**Reply by pinkfloydx33** [reply]
I can't think of a single time where I had two functions with the exact same body like this, particularly with both being single function invocations (other than explicit GetEnumerator maybe). What would be the point? Seems like it would make more sense to remove the duplicated method.

**Reply by cheong00** [reply]
> I can't think of a single time where I had two functions with the exact same body like this, particularly with both being single function invocations (other than explicit GetEnumerator maybe). What would be the point? Seems like it would make more sense to remove the duplicated method.

Well, sometimes we do that when need to implement methods in interface/abstract classes. However even in those cases I just group the code in a private method and uses expression-bodied methods approach above to code it.

**Reply by lakylekidd** [reply]
Yea, sounds logical. It would indeed not save too much code.. I myself, only found one use-case where it would help.
Thank you all for your input 😊
